tests/Functional/Schema/MySQLSchemaManagerTest.php

```diff
@@ -223,7 +223,7 @@ public function testColumnCollation(): void
 
         $columns = $this->schemaManager->listTableColumns('test_collation');
 
-        self::assertArrayNotHasKey('collation', $columns['id']->getPlatformOptions());
+        self::assertFalse($columns['id']->hasPlatformOption('collation'));
```

Comment:
Due to the code change, now platform options contain `['collation' => null]`, but this is an insignificant change. Ideally, `setPlatformOption()` should unset the key if the null value is passed and return `null` here if the option is not set: https://github.com/doctrine/dbal/blob/a45d5637e2ca92d0fc0bc050a638f7d8e85d1246/src/Schema/Column.php#L235-L238